test/unit/controlledGateway.test.ts

```diff
@@ -0,0 +1,166 @@
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/dist/src/signers';
+import {expect} from 'chai';
+import {ethers} from 'hardhat';
+import {
+  ControlledGateway,
+  ControlledGateway__factory,
+  LivepeerToken,
+  LivepeerToken__factory,
+} from '../../typechain';
+
+describe('L1 Gateway', function() {
+  let token: LivepeerToken;
+  let gateway: ControlledGateway;
+  let owner: SignerWithAddress;
+  let notOwner: SignerWithAddress;
+  let governor: SignerWithAddress;
+  let governor2: SignerWithAddress;
+
+  const ADMIN_ROLE =
+    '0x0000000000000000000000000000000000000000000000000000000000000000';
+
+  const GOVERNOR_ROLE = ethers.utils.solidityKeccak256(
+      ['string'],
+      ['GOVERNOR_ROLE'],
+  );
+
+  beforeEach(async function() {
+    const signers = await ethers.getSigners();
+    owner = signers[0];
+    governor = signers[1];
+    governor2 = signers[2];
+    notOwner = signers[3];
+
+    const Token: LivepeerToken__factory = await ethers.getContractFactory(
+        'LivepeerToken',
+    );
+    token = await Token.deploy();
+    await token.deployed();
+
+    const Gateway: ControlledGateway__factory = await ethers.getContractFactory(
+        'ControlledGateway',
+    );
+    gateway = await Gateway.deploy(token.address);
+    await gateway.deployed();
+  });
+
+  it('should correctly set admin', async function() {
+    const hasAdminRole = await gateway.hasRole(ADMIN_ROLE, owner.address);
+    expect(hasAdminRole).to.be.true;
+  });
+
+  describe('AccessControl', async function() {
+    describe('addgovernor', async function() {
+      describe('caller is not admin', async function() {
+        it('should not be able to set governance controller', async function() {
```

Comment:
```suggestion
        it('should not be able to set governor', async function() {
```

And let's update the rest of the terminology in the file as well.